src/poliastro/core/events.py

```diff
@@ -0,0 +1,65 @@
+import numpy as np
+from numba import njit as jit
+from numpy.linalg import norm
+
+from poliastro.core.elements import rv2coe
+
+
+@jit
+def continuous_shadow_function_for_penumbra(k, u_, r_sec, R_sec, R_primary):
+    p, ecc, inc, raan, argp, nu = rv2coe(k, u_[:3], u_[3:])
+
+    Px = np.cos(raan) * np.cos(argp) - np.sin(raan) * np.sin(argp) * np.cos(inc)
```

Comment:
This logic is already implemented in poliastro, see:

```python
def coe_rotation_matrix(inc, raan, argp):
    """Create a rotation matrix for coe transformation"""
    r = rotation_matrix(raan, 2)
    r = r @ rotation_matrix(inc, 0)
    r = r @ rotation_matrix(argp, 2)
    return r
```

where `P` and `Q` are the first and second column vectors of previous output matrix. Therefore, this logic could be simplified to:

```python
PQW = coe_rotation_matrix(inc, raan, argp)
P, Q = PQW[:,0], PQW[:,1]
```

Reply:
Oh, I forgot to use that. Thanks so much for this!

Reply:
Yeah, it was a little bit hidden...